src/main/scala/org/bblfsh/client/v2/BblfshClient.scala

```diff
@@ -218,21 +223,40 @@ object BblfshClient {
     Libuast.UastIterExt(node, treeOrder)
   }
 
-  /** Factory method for iterator over an managed node */
+  /** Factory method for iterator over a managed node */
   def iterator(node: JNode, treeOrder: TreeOrder): Libuast.UastIter = {
     Libuast.UastIter(node, treeOrder)
   }
 
-  /** Factory method for iterator over an native node, filtered by XPath query */
-  def filter(node: NodeExt, query: String):  Libuast.UastIterExt = Libuast.synchronized {
+  /** Factory method for iterator over a native node, filtered by XPath query */
+  def filter(node: NodeExt, query: String): Libuast.UastIterExt = Libuast.synchronized {
     node.filter(query)
   }
 
-  /** Factory method for iterator over an managed node, filtered by XPath query */
-  def filter(node: JNode, query: String):  Libuast.UastIter = Libuast.synchronized {
+  /** Factory method for iterator over a managed node, filtered by XPath query */
+  def filter(node: JNode, query: String): Libuast.UastIter = Libuast.synchronized {
     val ctx = Context()
     ctx.filter(query, node)
     // do not dispose the context, iterator steals it
   }
 
+  // Method to filter managed nodes based on type
+  // The aim is to not return a JNode but a specialized subclass of JNode, as JBool,
+  // for example, when T = JBool
```

Comment:
I suggest converting these comments to a ScalaDoc.

Reply:
Done!